src/lms5xx_node.cpp

```diff
@@ -0,0 +1,263 @@
+  scan_msg.angle_max = (double)output_range.stop_angle / 10000.0 * DEG2RAD - M_PI / 2;
+
+  multi_scan_msg.range_min = 0.01;
+  multi_scan_msg.range_max = 20.0; // TODO: value?
+  multi_scan_msg.scan_time = 100.0 / cfg.scan_frequency;
+  multi_scan_msg.angle_increment = (double)output_range.angular_resolution / 10000.0 * DEG2RAD;
+  multi_scan_msg.angle_min = (double)output_range.start_angle / 10000.0 * DEG2RAD - M_PI / 2;
+  multi_scan_msg.angle_max = (double)output_range.stop_angle / 10000.0 * DEG2RAD - M_PI / 2;
+
+  ROS_DEBUG_STREAM("Device resolution is " << (double)output_range.angular_resolution / 10000.0 << " degrees.");
+  ROS_DEBUG_STREAM("Device frequency is " << (double)cfg.scan_frequency / 100.0 << " Hz");
+
+  int angle_range = output_range.stop_angle - output_range.start_angle;
+  int num_values = angle_range / output_range.angular_resolution ;
+  if (angle_range % output_range.angular_resolution == 0)
+  {
+    // Include endpoint
+    ++num_values;
+  }
+  scan_msg.ranges.resize(num_values);
+  scan_msg.intensities.resize(num_values);
+
+  multi_scan_msg.ranges.resize(5);
+  multi_scan_msg.intensities.resize(5);
+  for (size_t i = 0; i < 5; ++i)
+  {
+    multi_scan_msg.ranges[i].echoes.resize(num_values);
+    multi_scan_msg.intensities[i].echoes.resize(num_values);
+  }
+
+  scan_msg.time_increment =
+      (output_range.angular_resolution / 10000.0)
+      / 360.0
+      / (cfg.scan_frequency / 100.0);
+
+  multi_scan_msg.time_increment =
+      (output_range.angular_resolution / 10000.0)
+      / 360.0
+      / (cfg.scan_frequency / 100.0);
+
+  ROS_DEBUG_STREAM("Time increment is " << static_cast<int>(scan_msg.time_increment * 1000000) << " microseconds");
+
+  dataCfg.output_channel = 5;
+  dataCfg.remission = true;
+  dataCfg.resolution = 0;
+  dataCfg.encoder = 0;
+  dataCfg.position = false;
+  dataCfg.device_name = false;
+  dataCfg.comment = false;
+  dataCfg.timestamp = false;
+  dataCfg.output_interval = 1;
+
+  ROS_DEBUG("Setting echo filter.");
+  laser.set_echo_filter(echo_mode);
+
+  ROS_DEBUG("Setting scan data configuration.");
+  laser.set_scan_data_config(dataCfg);
+
+  ROS_DEBUG("Saving configuration.");
+  laser.save_config();
+  return true;
+}
+
+int main(int argc, char **argv)
+{
+  if (argc == 2)
+  {
+    std::string arg(argv[1]);
+    if (arg == "-h" || arg == "--help")
+    {
+      usage();
+      return 0;
+    }
+  }
+
+  // laser data
+  LMS5xx laser;
+  sensor_msgs::LaserScan scan_msg;
+  sensor_msgs::MultiEchoLaserScan multi_scan_msg;
+
+  // parameters
+  std::string host;
+  std::string frame_id;
+  int port;
+  std::string echoes;
+  LMS5xx::EchoFilter echo_mode = LMS5xx::EchoFilter::AllEchoes;
+
+  ros::init(argc, argv, "lms5xx");
+  ros::NodeHandle nh;
+  ros::NodeHandle n("~");
+  ros::Publisher scan_pub = nh.advertise<sensor_msgs::LaserScan>("scan", 1);
+  ros::Publisher multi_pub;
+
+  n.param<std::string>("host", host, "192.168.0.1");
```

Comment:
We should produce a warning or even exit the program with an error_code if no host or port was set.